slog.go

```diff
@@ -0,0 +1,214 @@
+package btclog
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+)
+
+// Handler wraps the slog.Handler interface with a few more methods that we
+// need in order to satisfy the Logger interface.
+type Handler interface {
+	slog.Handler
+
+	// Level returns the current logging level of the Handler.
+	Level() Level
+
+	// SetLevel changes the logging level of the Handler to the passed
+	// level.
+	SetLevel(level Level)
+}
+
+// sLogger is an implementation of Logger backed by a structured sLogger.
+type sLogger struct {
+	Handler
+	logger *slog.Logger
+}
+
+// NewSLogger constructs a new structured logger from the given Handler.
+func NewSLogger(handler Handler) Logger {
```

Comment:
nit: I think the general rule is "accept interfaces, return concrete types", so I think this should return `*sLogger` instead? Or does the linter then complain about exported function with un-exported type?

Reply:
No strong opinion, more a rule of thumb.